rosbag2_storage_sqlite3/include/rosbag2_storage_sqlite3/sqlite_storage.hpp

```diff
@@ -138,9 +144,11 @@ class ROSBAG2_STORAGE_DEFAULT_PLUGINS_PUBLIC SqliteStorage
   ReadQueryResult message_result_ {nullptr};
   ReadQueryResult::Iterator current_message_row_ {
     nullptr, SqliteStatementWrapper::QueryResult<>::Iterator::POSITION_END};
-  std::unordered_map<std::string, int> topics_ RCPPUTILS_TSA_GUARDED_BY(db_read_write_mutex_);
-  std::unordered_map<std::string, int> msg_definitions_ RCPPUTILS_TSA_GUARDED_BY(
+  std::unordered_map<std::string, int64_t> topics_ RCPPUTILS_TSA_GUARDED_BY(db_read_write_mutex_);
+  std::unordered_map<std::string, int64_t> msg_definitions_ RCPPUTILS_TSA_GUARDED_BY(
     db_read_write_mutex_);
+  std::unordered_map<int64_t, uint16_t> inner_to_extern_topic_id_map_;
+  uint16_t last_extern_topic_id_ = 0;  // 0 corresponds to the invalid topic_id
```

Comment:
how about the following?

```suggestion
  std::atomic<uint16_t> last_extern_topic_id_(0);  // 0 corresponds to the invalid topic_id
```